src/main/java/org/opensearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -266,21 +268,32 @@ public KNNVectorFieldMapper build(BuilderContext context) {
     }
 
     public static class TypeParser implements Mapper.TypeParser {
+
+        // Use a supplier here because in {@link org.opensearch.knn.KNNPlugin#getMappers()} the ModelDao has not yet
+        // been initialized
+        private Supplier<ModelDao> modelDaoSupplier;
+
+        public TypeParser(Supplier<ModelDao> modelDaoSupplier) {
+            super();
```

Comment:
Looks like TypeParser only implements not extends. Do we need super?

Reply:
Ah good catch. No we do not